api/discovery/discovery.go

```diff
@@ -52,7 +57,12 @@ type Authorizer interface {
 }
 
 // NewResource creates a new voc resource.
-func NewResource(d Discoverer, ID voc.ResourceID, name string, creationTime *time.Time, location voc.GeoLocation, labels map[string]string, typ []string) *voc.Resource {
+func NewResource(d Discoverer, ID voc.ResourceID, name string, creationTime *time.Time, location voc.GeoLocation, labels map[string]string, typ []string, raw map[string][]interface{}) *voc.Resource {
```

Comment:
Almost there... just wondering... Why is "raw" a map and not just a list? If it would be a list, we could just use varargs and use it like `NewResource(..., myAzureResource, myOtherResource)` and avoid the `raw := make map` stuff.

Reply:
I thought it is better if we have a map with the object type as a key string. Otherwise it is difficult to understand (depending on the cloud) which entry contains the raw data for which object type.